src/ngAnimate/animateQueue.js

```diff
@@ -59,11 +67,38 @@ var $$AnimateQueueProvider = ['$animateProvider', function($animateProvider) {
   });
 
   rules.cancel.push(function(element, newAnimation, currentAnimation) {
-    var nO = newAnimation.options;
-    var cO = currentAnimation.options;
+    var ONE_SPACE = ' ';
+
+    var nA = newAnimation.options.addClass;
+    var nR = newAnimation.options.removeClass;
+    var cA = currentAnimation.options.addClass;
+    var cR = currentAnimation.options.removeClass;
+
+    // early detection to save the global CPU shortage :)
+    if ((!isDefined(nA) && !isDefined(nR)) || (!isDefined(cA) && !isDefined(cR))) {
+      return false;
+    }
+
+    var cancelSomething = false;
+
+    cA = cA ? makeTruthyMapFromKeys(cA.split(ONE_SPACE)) : null;
+    cR = cR ? makeTruthyMapFromKeys(cR.split(ONE_SPACE)) : null;
```

Comment:
This repeated logic here could be moved into the `makeTrutyMapFromKeys` and perhaps be renamed to `makeCssClassMap`